packages/carbon-web-components/src/components/pagination-nav/pagination-nav.ts

```diff
@@ -0,0 +1,323 @@
+  totalItems = 1;
+
+  /**
+   * Array allowing us to map out the items
+   *   - Prefer showing this.count
+   *   - Maximum of this.visible
+   *   - Minumum of 1
+   */
+  private _iterator: any[] = [];
+
+  shouldUpdate(changedProperties) {
+    // Prevent setting "itemsShown" to less than 5.
+     if (changedProperties.has('itemsShown') && this.itemsShown < 5) {
+      this.itemsShown = 5;
+    }
+
+    // Prevent setting "page" outside bounds of available pages.
+    if (changedProperties.has('count') || changedProperties.has('page')) {
+      if (this.page > this.totalItems) {
+        this.page = this.totalItems - 1;
+      }
+      if (this.page < 0) {
+        this.page = 0;
+      }
+    }
+
+    return true;
+  }
+
+  /**
+   * After values are updated, but before rendering, update the iterator.
+   *
+   * @param changedProperties map of previous values of changed properties
+   */
+  update(changedProperties) {
+    if (changedProperties.has('totalItems') || changedProperties.has('itemsShown') || changedProperties.has('page')) {
+      const { totalItems, itemsShown: visibleLimit, page } = this;
+      const { calculateCuts } = (this.constructor as typeof BXPaginationNav);
+      const allItems = [...new Array(Math.max(totalItems, 1))].map((_u, i) => i);
+      let iterator;
+
+      if (allItems.length > visibleLimit) {
+        const first = allItems.shift();
+        const last = allItems.pop();
+        const {front, back} = calculateCuts(page, allItems.length, visibleLimit - 2);
+        const upperGroup = allItems.splice(back);
+        const lowerGroup = allItems.splice(0, front);
+        const upper = upperGroup.length > 1 ? upperGroup : upperGroup[0];
+        const lower = lowerGroup.length > 1 ? lowerGroup : lowerGroup[0]
+
+        iterator = [first, lower, ...allItems, upper, last]
+          .filter(item => (typeof item === "number" || (Array.isArray(item) && item.length)));
+      } else {
+        iterator = allItems;
+      }
+
+      this._iterator = iterator;
+    }
+    super.update(changedProperties);
+  }
+
+  /**
+   * After page is selected, dispatch event & ensure focus is on active page.
+   *
+   * @param changedProperties map of changed props.
+   */
+  updated(changedProperties) {
+    if (changedProperties.has('page')) {
+      const { pageChangedEvent } = (this.constructor as typeof BXPaginationNav);
+      const event = new CustomEvent(pageChangedEvent, {
+        bubbles: true,
+        cancelable: true,
+        composed: true,
+        detail: {
+          oldValue: changedProperties.get('page'),
+          newValue: this.page,
+        }
+      })
+      this.dispatchEvent(event);
+      window.requestIdleCallback(()=>{
+        const activeItem = (this.shadowRoot!.querySelector('.bx--pagination-nav__page--active') as HTMLElement);
+        if (activeItem && !activeItem.matches(':focus')) {
+          activeItem.focus();
+        }
+      })
```

Comment:
Though this has the desired effect when activating an item, it also steals focus when the component is mounted. Also, `requestIdleCallback` isn't yet supported in Safari – https://caniuse.com/requestidlecallback.